packages/itmat-commons/src/graphql/files.ts

```diff
@@ -1,15 +1,16 @@
 import gql from 'graphql-tag';
 
 export const UPLOAD_FILE = gql`
-    mutation uploadFile($studyId: String!, $file: Upload!, $description: String!, $fileLength: Int) {
-        uploadFile(studyId: $studyId, description: $description, file: $file, fileLength: $fileLength) {
+    mutation uploadFile($studyId: String!, $file: Upload!, $description: String!, $fileLength: Int, $hash: String!) {
+        uploadFile(studyId: $studyId, description: $description, file: $file, fileLength: $fileLength, hash: $hash) {
```

Comment:
I'm not sure I understand why you adopted the stragegy of passing the hash to the upload function.
This requires the has to be calculated by the client-side prior to upload. Is that what is expected ?